src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/diagnostics/support/EventListenersComponent.java

```diff
@@ -0,0 +1,138 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.support;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.PrintedContent;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.Messages;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.Diagnostics;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.EventListenersReport;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.EventListener;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.GerritTrigger;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.data.Branch;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.data.GerritProject;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.events.PluginGerritEvent;
+import com.sonymobile.tools.gerrit.gerritevents.GerritEventListener;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Job;
+import hudson.security.Permission;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Support Core Component writing the {@link EventListenersReport}.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+@Extension(optional = true)
+public class EventListenersComponent extends Component {
+
+    /**
+     * The max number of interesting Project or branch patterns to list for a job.
+     */
+    private static final int MAX_PROJECT_BRANCH_LIST_LENGTH = 5;
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Diagnostics.getRequiredPermission());
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Gerrit: " + Messages.EventListenersReport_DisplayName();
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        container.add(new PrintedContent("gerrit/event-listeners.md") {
```

Comment:
if this is going to be producing lots of content it should not be enabled by default... or else find some better way to summarize